extension/runtime/src/main/java/io/quarkiverse/temporal/WorkerFactoryRecorder.java

```diff
@@ -50,9 +51,9 @@ public WorkerOptions createWorkerOptions(WorkerRuntimeConfig config) {
                 .setStickyQueueScheduleToStartTimeout(config.stickyQueueScheduleToStartTimeout())
                 .setDisableEagerExecution(config.disableEagerExecution())
                 .setUseBuildIdForVersioning(config.useBuildIdForVersioning())
-                .setStickyTaskQueueDrainTimeout(config.stickyTaskQueueDrainTimeout());
+                .setStickyTaskQueueDrainTimeout(config.stickyTaskQueueDrainTimeout())
+                .setBuildId(config.buildId().orElseGet(() -> CDI.current().select(GitInfo.class).get().latestCommitId()));
```

Comment:
I am wondering should it fail back to the Infor plugin application ID or the GitCommit is not available?

Reply:
I am not sure I understood the question, what should it fail back to ? Is the application id something generated by quarkus ?

Reply:
so its the application version ID like "10.2.4" so if git commit is not available use the application version as a fallback?

Reply:
hum I guess this is okay-ish, but I can imagine multiple scenario in which this would be a pretty bad identifier:
- in case of a dev version (snapshots all share the same version but the code is different)
- one could delete an artifact and rebuild it with the same version number

Reply:
Understood. Let's go with this until someone thinks it's not good enough!